libpod/inspect_data.go

```diff
@@ -0,0 +1,204 @@
+	Mounts          []specs.Mount    `json:"Mounts"`
+	Config          *CtrConfig       `json:"Config"`
+	NetworkSettings *NetworkSettings `json:"NetworkSettings"` // not implemented
+}
+
+// LogConfig holds the log information for a container
+type LogConfig struct {
+	Type   string            `json:"Type"`   // not implemented
+	Config map[string]string `json:"Config"` //idk type, not implemented
+}
+
+// RestarPolicy holds the container's policy on restart
+type RestartPolicy struct {
+	Name              string `json:"Name"`              // not implemented
+	MaximumRetryCount int    `json:"MaximumRetryCount"` //check type, not implemented
+}
+
+// CtrState represents the state of a container.
+type CtrState struct {
+	OciVersion string    `json:"OciVersion"`
+	Status     string    `json:"Status"`
+	Running    bool      `json:"Running"`
+	Paused     bool      `json:"Paused"`
+	Restarting bool      `json:"Restarting"` // not implemented
+	OOMKilled  bool      `json:"OOMKilled"`
+	Dead       bool      `json:"Dead"`
+	Pid        int       `json:"Pid"`
+	ExitCode   int32     `json:"ExitCode"`
+	Error      string    `json:"Error"` // not implemented
+	StartedAt  time.Time `json:"StartedAt"`
+	FinishedAt time.Time `json:"FinishedAt"`
+}
+
+// HostConfig represents the host configuration for the container
+type HostConfig struct {
+	Binds                []string                    `json:"Binds"`
+	ContainerIDFile      string                      `json:"ContainerIDFile"` //not implemented
+	LogConfig            *LogConfig                  `json:"LogConfig"`       //not implemented
+	NetworkMode          string                      `json:"NetworkMode"`     //not implemented
+	PortBindings         map[string]struct{}         `json:"PortBindings"`    //not implemented
+	RestartPolicy        *RestartPolicy              `json:"RestartPolicy"`   //not implemented
+	AutoRemove           bool                        `json:"AutoRemove"`      //not implemented
+	VolumeDriver         string                      `json:"VolumeDriver"`    //not implemented
+	VolumesFrom          string                      `json:"VolumesFrom"`     // dont know type, not implemented
+	CapAdd               string                      `json:"CapAdd"`          //dont know type, not implemented
+	CapDrop              string                      `json:"CapDrop"`         //dont know type, not implemented
+	Dns                  []string                    `json:"Dns"`             //not implemented
+	DnsOptions           []string                    `json:"DnsOptions"`      //not implemented
+	DnsSearch            []string                    `json:"DnsSearch"`       //not implemented
+	ExtraHosts           string                      `json:"ExtraHosts"`      //dont know type, not implemented
+	GroupAdd             string                      `json:"GroupAdd"`        //not implemented
+	IpcMode              string                      `json:"IpcMode"`         //not implemented
```

Comment:
Can probably be parsed out of the spec's namespaces info

Reply:
fixed